packages/autocomplete-plugin-recent-searches/src/createPlugin.ts

```diff
@@ -0,0 +1,66 @@
+import { AutocompletePlugin } from '@algolia/autocomplete-core';
+
+import { createRecentSearchStore } from './createRecentSearchStore';
+
+type PluginOptions = {
+  limit?: number;
+};
+
+type RecentSearchesPlugin<TItem> = AutocompletePlugin<TItem> & {
+  getFacetFilters: () => string[];
+};
+
+export function createPlugin<TItem>({
+  limit = 5,
+}: PluginOptions = {}): RecentSearchesPlugin<TItem> {
+  const store = createRecentSearchStore({
+    key: 'RECENT_SEARCHES',
+    limit,
+  });
+
+  return {
+    getSources: ({ query }) => {
+      return query
+        ? []
+        : [
+            {
+              getInputValue: ({ suggestion }) => suggestion.query,
+              getSuggestions() {
+                return store.getAll();
+              },
+              onSelect({ suggestion }) {
+                store.add(suggestion);
+              },
+              templates: {
+                item({ item }) {
+                  return `
+                <div class="autocomplete-item">
+                  <div>
+                    <svg viewBox="0 0 22 22" width="16" height="16" fill="currentColor">
+                      <path d="M0 0h24v24H0z" fill="none"/>
+                      <path d="M13 3c-4.97 0-9 4.03-9 9H1l3.89 3.89.07.14L9 12H6c0-3.87 3.13-7 7-7s7 3.13 7 7-3.13 7-7 7c-1.93 0-3.68-.79-4.94-2.06l-1.42 1.42C8.27 19.99 10.51 21 13 21c4.97 0 9-4.03 9-9s-4.03-9-9-9zm-1 5v5l4.28 2.54.72-1.21-3.5-2.08V8H12z"/>
+                    </svg>
+
+                    <span>${item.query}</span>
+                  </div>
+                </div>
+              `;
+                },
+              },
+            },
+          ];
+    },
+    onSubmit: ({ state }) => {
+      store.add({
+        objectID: state.query,
+        query: state.query,
+      });
+    },
+    onSelect: ({ suggestion }) => {
+      store.add(suggestion);
+    },
+    getFacetFilters: () => {
```

Comment:
I wonder if custom plugin methods should be stored in a key, otherwise people will start storing plenty of stuff at the top-level, making the plugin API less consistent.

Reply:
So far I only can think of having a namespace there for custom plugin methods. 

Or,

```js
const [plugin, { getFacetFilters, someOtherMethod }] = createRecentSearchesPlugin({});
```

but I'm not a fan. Any idea?

Reply:
@francoischalifour I feel like once we agree with something on this part, we can merge this now as a minimal ground work and start creating new PRs for many features on top of it. What do you think?

Reply:
For now we can have like: `plugin.data.getFacetFilters`.

(The word `data` is bad but it's to make sure to find a better one before releasing.)

Reply:
I think it'd make more sense to have:

```ts
AutocompletePlugin<TItem, TData>
```

Reply:
https://github.com/algolia/autocomplete.js/pull/316/commits/6446097a7d4113bc25292773a39f303739ace96d